src/vmshepherd/errors.py

```diff
@@ -0,0 +1,58 @@
+class VmShepherdError(Exception):
+    message = None
+    details = None
+
+    def __init__(self, message, details=None):
+        super().__init__(self.message)
+        self.message = message
+        self.details = details if details else ''
```

Comment:
IMO there is no point in this `if` if we apply default value to the `details` kwarg 
```suggestion
        self.details = details
```